src/TConsole.cpp

```diff
@@ -2552,6 +2552,9 @@ void TConsole::slot_searchBufferUp()
     if (_txt != mSearchQuery) {
         mSearchQuery = _txt;
         mCurrentSearchResult = buffer.lineBuffer.size();
+    } else {
+        // make sure the line to search from does not exceed the buffer, which can grow and shrink dynamically
+        mCurrentSearchResult = std::min(mCurrentSearchResult, buffer.lineBuffer.size());
```

Comment:
Is there any preference project-wide whether to use `std::min` or `qMin`?

Reply:
I'm a fan of using the standard library if something is in there, otherwise we look to Qt